### PR TITLE
Update `load_fixtures.sh` and create `create_fixtures.sh`

### PR DESCRIPTION
**Pull request by jacobdgm**
This PR updates a script used to load data into the database and creates a new script to make it simpler to export data from a running instance of CantusDB.

It updates `load_fixtures.sh`, by:
- removing `indexer_fixture.json` from the list of fixtures to import (we no longer have an indexer model)
- adding `article_fixture.json` (this had not been updated following the introduction of the `articles` app and the `Article` model)

It adds a new file, `create_fixtures.sh`. If one is exporting data from an existing instance of CantusDB, this script can be used to ensure all the fixtures necessary for running `load_fixtures.sh` are created.

(Hopefully, once we have all the instances of CantusDB running the same version of postgres, these changes will be somewhat moot, since we'll be able to export and install the postgres database _in toto_, but the changes in this PR will at least ensure that the "The beaten path (done many times, surely works)" detailed in the [Wiki](https://github.com/DDMAL/CantusDB/wiki/Getting-Started-with-Development) are up-to-date and working.)

(Note to future Jacob: It seems that the change to `docker-compose.yml` came along for the ride too, so probably best to wait on merging this PR until #590 has been merged.)

### CODE REVIEW

**Review by dchiller — APPROVED** [review]

**Comment by jacobdgm** [comment]
ugh... I ran this updated version of `load_fixtures.sh` this morning - it's just finished, and it seems to have imported about 1/3 of the chants that are present on staging (I used fixtures taken from staging).

I was getting lots of tracebacks and error messages while it was working through the numerous `chant_fixture_NNNN`s. Something like this happened when I first set up CantusDB locally a year ago (I recall being alarmed that there were lots of error messages being printed to the terminal, and being somewhat distressed when I was told to ignore them - the data ended up being loaded in just fine), so I was curious to see whether this was the same thing - evidently not. Here's the tail end of what was printed to my terminal while running the script:

<details>
<pre>
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/code/django/cantusdb_project/manage.py", line 32, in <module>
    main()
  File "/code/django/cantusdb_project/manage.py", line 28, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 440, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 402, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 102, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 171, in loaddata
    connection.check_constraints(table_names=table_names)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 314, in check_constraints
    cursor.execute("SET CONSTRAINTS ALL IMMEDIATE")
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
django.db.utils.IntegrityError: Problem installing fixtures: insert or update on table "main_app_chant" violates foreign key constraint "main_app_chant_next_chant_id_2ace4c77_fk_main_app_chant_id"
DETAIL:  Key (next_chant_id)=(248845) is not present in table "main_app_chant".

Loading './main_app/fixtures/chant_fixtures/chant_fixture_997' fixtures...
Checking '/code/django/cantusdb_project/main_app/fixtures/./main_app/fixtures/chant_fixtures' for fixtures...
No fixture 'chant_fixture_997' in '/code/django/cantusdb_project/main_app/fixtures/./main_app/fixtures/chant_fixtures'.
Checking '/code/django/cantusdb_project/./main_app/fixtures/chant_fixtures' for fixtures...
Installing json fixture 'chant_fixture_997' from '/code/django/cantusdb_project/./main_app/fixtures/chant_fixtures'.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
psycopg2.errors.ForeignKeyViolation: insert or update on table "main_app_chant" violates foreign key constraint "main_app_chant_next_chant_id_2ace4c77_fk_main_app_chant_id"
DETAIL:  Key (next_chant_id)=(248945) is not present in table "main_app_chant".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/code/django/cantusdb_project/manage.py", line 32, in <module>
    main()
  File "/code/django/cantusdb_project/manage.py", line 28, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 440, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 402, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 102, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 171, in loaddata
    connection.check_constraints(table_names=table_names)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 314, in check_constraints
    cursor.execute("SET CONSTRAINTS ALL IMMEDIATE")
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
django.db.utils.IntegrityError: Problem installing fixtures: insert or update on table "main_app_chant" violates foreign key constraint "main_app_chant_next_chant_id_2ace4c77_fk_main_app_chant_id"
DETAIL:  Key (next_chant_id)=(248945) is not present in table "main_app_chant".

Loading './main_app/fixtures/chant_fixtures/chant_fixture_998' fixtures...
Checking '/code/django/cantusdb_project/main_app/fixtures/./main_app/fixtures/chant_fixtures' for fixtures...
No fixture 'chant_fixture_998' in '/code/django/cantusdb_project/main_app/fixtures/./main_app/fixtures/chant_fixtures'.
Checking '/code/django/cantusdb_project/./main_app/fixtures/chant_fixtures' for fixtures...
Installing json fixture 'chant_fixture_998' from '/code/django/cantusdb_project/./main_app/fixtures/chant_fixtures'.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
psycopg2.errors.ForeignKeyViolation: insert or update on table "main_app_chant" violates foreign key constraint "main_app_chant_next_chant_id_2ace4c77_fk_main_app_chant_id"
DETAIL:  Key (next_chant_id)=(249045) is not present in table "main_app_chant".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/code/django/cantusdb_project/manage.py", line 32, in <module>
    main()
  File "/code/django/cantusdb_project/manage.py", line 28, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 440, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 402, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 102, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 171, in loaddata
    connection.check_constraints(table_names=table_names)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 314, in check_constraints
    cursor.execute("SET CONSTRAINTS ALL IMMEDIATE")
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
django.db.utils.IntegrityError: Problem installing fixtures: insert or update on table "main_app_chant" violates foreign key constraint "main_app_chant_next_chant_id_2ace4c77_fk_main_app_chant_id"
DETAIL:  Key (next_chant_id)=(249045) is not present in table "main_app_chant".

Loading './main_app/fixtures/chant_fixtures/chant_fixture_999' fixtures...
Checking '/code/django/cantusdb_project/main_app/fixtures/./main_app/fixtures/chant_fixtures' for fixtures...
No fixture 'chant_fixture_999' in '/code/django/cantusdb_project/main_app/fixtures/./main_app/fixtures/chant_fixtures'.
Checking '/code/django/cantusdb_project/./main_app/fixtures/chant_fixtures' for fixtures...
Installing json fixture 'chant_fixture_999' from '/code/django/cantusdb_project/./main_app/fixtures/chant_fixtures'.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
psycopg2.errors.ForeignKeyViolation: insert or update on table "main_app_chant" violates foreign key constraint "main_app_chant_next_chant_id_2ace4c77_fk_main_app_chant_id"
DETAIL:  Key (next_chant_id)=(249145) is not present in table "main_app_chant".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/code/django/cantusdb_project/manage.py", line 32, in <module>
    main()
  File "/code/django/cantusdb_project/manage.py", line 28, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 440, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 402, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 102, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 171, in loaddata
    connection.check_constraints(table_names=table_names)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/postgresql/base.py", line 314, in check_constraints
    cursor.execute("SET CONSTRAINTS ALL IMMEDIATE")
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 102, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
django.db.utils.IntegrityError: Problem installing fixtures: insert or update on table "main_app_chant" violates foreign key constraint "main_app_chant_next_chant_id_2ace4c77_fk_main_app_chant_id"
DETAIL:  Key (next_chant_id)=(249145) is not present in table "main_app_chant"</pre>
</details>

The new problem is perhaps the `django.db.utils.IntegrityError` - we have a field, `next_chant`, that points to another `Chant` in the database. My best guess is that whenever we try to add a chant to the database whose `next_chant` points to a chant that hasn't yet been imported, the new chant doesn't get added. Maybe the entire 100-chant fixture fails with it? 

I'll look into this in more detail next week. Brainstorming solutions:
- There's probably a django-native way to deal with this situation (though perhaps they won't work because the chant fixture is so big that we've chosen to break it into a bunch of small fixtures). Failing this, perhaps we can `sed` through all the created fixtures and remove this item from every chant in the fixture - we have a script to populate chants' `next_chant` fields anyways. 
- Or, we can abandon the fixtures approach entirely and use @jwang44's newer approach of exporting the entire database (see ["the faster path"](https://github.com/DDMAL/CantusDB/wiki/Getting-Started-with-Development#loading-initial-data-two-methods)). We'll need to figure out how to update the database version, but this should be doable.
  - related/similar to this, on a single machine running the proper version of Postgres (starting with a clean version of the database if necessary), we could sync data from OldCantus, and once this is done, we can export the entire database. On any other machine, we can ensure we're working with a blank slate in the proper version of Postgres, and plug this exported database into that CantusDB app. This would also give us a chance to do a dry run of the data-syncing process well in advance of doing the switchover from OldCantus to New.

**Comment by jacobdgm** [comment]
tl;dr: `load_fixtures.sh` doesn't work. It's probably because CantusDB's chants have a field, `next_chant`, that wasn't present when `load_fixtures.sh` was written.

**Comment by dchiller** [comment]
Just looking at the traceback and doing some googling and I suspect your suspicion about the cause is correct. 

A few other thoughts:
1. I feel like it would make sense to backup the database using the second method in addition to the first. Since as I understand them the fixtures are serializing tables in the db one by one it would make sense to me for the purposes of having a good archive of the entire database.
2. For the purposes of upgrading versions of Postgres, it seems like more recent versions of Postgres offer a nice utility for that, and at the very least it would be worth checking out this page (https://www.postgresql.org/docs/current/upgrading.html). There's also some interesting and potentially useful tidbits (eg: "It is recommended that you use the pg_dump and pg_dumpall programs from the newer version of PostgreSQL, to take advantage of enhancements that might have been made in these programs.") 
3. Finally, it looks like for the fixtures route, this page would be useful (https://docs.djangoproject.com/en/4.0/topics/serialization/), especially the bits on forward references and dependencies during serialization...it looks like these mostly deal with foreign key relations, so not sure how they will translate when the references are "domestic" as in this case. I would be wary of doing some other sort of modification of the fixtures in the interim -- it feels like a recipe for having something happen that destroys a needed backup! (but I would be less concerned about it if both methods were used per 1 above).

**Comment by jacobdgm** [comment]
What do we think - should this pull request be closed, because the `load_fixtures.sh` script doesn't work? Should we merge it, because it's closer to working than it was before (e.g. we're not going to have an indexer fixture, because we no longer have an indexer table in the database)?

**Comment by jwang44** [comment]
I think merging it makes sense. Just add a note somewhere saying that it needs to be updated to address the `next_chant` field.
